src/components/lightbox/CaptionItem.tsx

```diff
@@ -0,0 +1,73 @@
+import { Button, Group, Stack, Textarea, Title } from "@mantine/core";
+
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import "react-virtualized/styles.css";
+import { Note } from "tabler-icons-react";
+import { useAppDispatch, useAppSelector } from "../../store/store";
+import { generatePhotoIm2txtCaption, editPhoto } from "../../actions/photosActions";
+
+type Props = {
+  isPublic: boolean;
+  photoDetail: any;
+  dispatch: any;
```

Comment:
instead of passing a dispatch function you can use the dispatch hook we provide [here](https://github.com/LibrePhotos/librephotos-frontend/blob/dev/src/store/store.ts#L29)

you can call it like

```
export function CaptionItem(props: Props) {
  const dispatch = useAppDispatch()
  const { generatingCaptionIm2txt } = useAppSelector(store => store.photos);
  const { photoDetail, isPublic, dispatch } = props;
  const { t } = useTranslation();
...
```